compositor_pipeline/src/pipeline/whip_whep/handlers.rs

```diff
@@ -0,0 +1,256 @@
+use std::sync::{Arc, Mutex};
+
+use crate::{
+    pipeline::{
+        input::whip::depayloader::Depayloader,
+        whip_whep::{
+            init_peer_connection,
+            utils::{authorize_token, handle_track},
+            WhipWhepState,
+        },
+        EncodedChunk,
+    },
+    queue::PipelineEvent,
+};
+use axum::{
+    body::Body,
+    extract::{Path, State},
+    http::{HeaderMap, Response, StatusCode},
+    response::IntoResponse,
+};
+use compositor_render::InputId;
+use serde_json::{json, Value};
+use tokio::sync::mpsc::Sender;
+use tracing::{debug, error, info};
+use webrtc::{
+    ice_transport::ice_candidate::RTCIceCandidateInit,
+    peer_connection::{sdp::session_description::RTCSessionDescription, RTCPeerConnection},
+    rtp_transceiver::rtp_codec::RTPCodecType,
+};
+
+use super::WhipServerError;
+
+pub async fn status() -> Result<(StatusCode, axum::Json<Value>), WhipServerError> {
+    info!("[status] got request");
+    Ok((StatusCode::OK, axum::Json(json!({}))))
+}
+
+#[axum::debug_handler]
+pub async fn handle_whip(
+    Path(id): Path<String>,
+    State(state): State<Arc<WhipWhepState>>,
+    headers: HeaderMap,
+    offer: String,
+) -> Result<Response<Body>, WhipServerError> {
+    info!("[whip] got headers: {headers:?}");
+    info!("[whip] got request: {offer}");
+
+    let input_id = InputId(Arc::from(id.clone()));
+
+    // Validate that the Content-Type is `application/sdp`
+    if let Some(content_type) = headers.get("Content-Type") {
+        if content_type.as_bytes() != b"application/sdp" {
+            error!("Invalid Content-Type, expecting application/sdp");
+            return Err(WhipServerError::InternalError(
```

Comment:
Isn't it a `BAD_REQUEST`? It's users responsibility to provide required header type.